src/Features/Core/Portable/Diagnostics/IDiagnosticService.cs

```diff
@@ -24,20 +25,47 @@ internal interface IDiagnosticService
         /// <summary>
         /// Get current diagnostics stored in IDiagnosticUpdateSource.
         /// </summary>
-        /// <param name="forPullDiagnostics">If the caller of this method will be using the diagnostics for pull or push
-        /// diagnostic purposes.  The <see cref="IDiagnosticService"/> only provides diagnostics for either push or pull
-        /// purposes (but not both).  If the caller's desired purpose doesn't match the service's setting, then this
-        /// will return nothing, otherwise it will return the requested diagnostics.</param>
-        ImmutableArray<DiagnosticData> GetDiagnostics(Workspace workspace, ProjectId? projectId, DocumentId? documentId, object? id, bool includeSuppressedDiagnostics, bool forPullDiagnostics, CancellationToken cancellationToken);
```

Comment:
ooh another "fun" complication - TS uses this API!  https://devdiv.visualstudio.com/DevDiv/_git/TypeScript-VS?path=%2FVS%2FLanguageService%2FTypeScriptLanguageService%2FLSP%2FCodeActions%2FQuickFixService.cs&version=GBmain&line=75&lineEnd=76&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

We may have to keep this around :(

Reply:
:(  this absolutely was not an API other teams were supposed to use :(  @DanielRosenwasser

Reply:
You should just ping @minestarks @uniqueiniquity, @jessetrinity as well - I have very little involvement/experience on the Roslyn integration side.

Reply:
Is there an alternative? Should we convert this to external access pattern?

Reply:
i've added this method back in for you guys for the time being.  talking to @uniqueiniquity you also plan to move to LSP at some poitn.  So we could ditch all of this once you have done that :)